x-pack/plugins/security_solution/server/fleet_integration/handlers/create_event_filters.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import uuid from 'uuid';
+import { i18n } from '@kbn/i18n';
+import { ENDPOINT_EVENT_FILTERS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import type { Logger } from '@kbn/core/server';
+import type { PackagePolicy } from '@kbn/fleet-plugin/common';
+import type { ExceptionListClient } from '@kbn/lists-plugin/server';
+import type { EventFilters } from '../types';
+
+const PROCESS_INTERACTIVE_ECS_FIELD = 'process.entry_leader.interactive';
+
+/**
+ * Check if Event Filter list exists and Create Event Filters for the Elastic Defend integration.
+ */
+export const createEventFilters = async (
+  logger: Logger,
+  exceptionsClient: ExceptionListClient,
+  eventFilters: EventFilters,
+  packagePolicy: PackagePolicy
+) => {
+  const exceptionList = await exceptionsClient.getExceptionList({
```

Comment:
Suggestion: Create the list and ignore `409` errors (409 is returned if teh list already exists. This is what we do in the UI side, so no issues if done here)

The list definition used via API is defined here:

https://github.com/elastic/kibana/blob/6efef0496077f4e61d49e4e43f75941ff3d98d9e/x-pack/plugins/security_solution/public/management/pages/event_filters/constants.ts#L22-L28

The service method to use would be `exceptionsClient.createExceptionLIst()` - you can't just reuse the list definition I reference above because its props are not camelCased and those are for use with the API not the service.  The good thing is that you get just about all that you need from exported `constants`.  The Exceptions List service has a `createTrustedAppsList()` method that you can use as reference. The call will probably look something like this:


```typescript
//...

import {
  ENDPOINT_EVENT_FILTERS_LIST_ID,
  ENDPOINT_EVENT_FILTERS_LIST_NAME,
  ENDPOINT_EVENT_FILTERS_LIST_DESCRIPTION,
} from '@kbn/securitysolution-list-constants';
import { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
import { SavedObjectsErrorHelpers } from '@kbn/core/server';

//.....

try {
  await exceptionsClient.createExceptionList({
    name: ENDPOINT_EVENT_FILTERS_LIST_NAME,
    namespaceType: 'agnostic',
    description: ENDPOINT_EVENT_FILTERS_LIST_DESCRIPTION,
    listId: ENDPOINT_EVENT_FILTERS_LIST_ID,
    type: ExceptionListTypeEnum.ENDPOINT_EVENTS,
    immutable: false,
    meta: undefined,
    tags: [],
    version: 1,
  });
} catch (err) {
  // Ignore 409
  if (!SavedObjectsErrorHelpers.isConflictError(err)) {
    throw err;
  }
}

```

Reply:
That was really helpful, thank you!